### PR TITLE
New registration instructions, MSAL Update and minor code refactor

### PR DESCRIPTION
**Pull request by titusgicheru**
- New Azure App registration instructions
- Updated to use MSAL
- Minor code update to accommodate the above

### CODE REVIEW

File: PeoplePicker.html [hunk 1]
```diff
@@ -1,230 +1,183 @@
...
+            var PeoplePickerElements = document.querySelectorAll(".ms-PeoplePicker");
+            for (var i = 0; i < PeoplePickerElements.length; i++) {
+              new fabric['PeoplePicker'](PeoplePickerElements[i]);
+            }
+          </script>       
+        
+          <!-- Authenticate with Office JS Helper -->
+          <script>
+            var clientId = "ae0b221d-6c54-4358-a365-6fb17d497305";
+            var redirectUrl = "https://localhost/PeoplePicker.html";
```
Comment (thewahome-zz) [inline PeoplePicker.html]: Here you could use window.location as the redirect uri. Keeps it clean.

Comment (thewahome-zz) [reply PeoplePicker.html]: Just seen its done in the next line 117, so removing this line is better

File: PeoplePicker.html [hunk 2]
```diff
@@ -1,230 +1,183 @@
...
+          <script type="text/javascript">
+            var PeoplePickerElements = document.querySelectorAll(".ms-PeoplePicker");
+            for (var i = 0; i < PeoplePickerElements.length; i++) {
+              new fabric['PeoplePicker'](PeoplePickerElements[i]);
+            }
+          </script>       
+        
+          <!-- Authenticate with Office JS Helper -->
+          <script>
+            var clientId = "ae0b221d-6c54-4358-a365-6fb17d497305";
```
Comment (thewahome-zz) [inline PeoplePicker.html]: Client id could be abstracted in a secrets.js file where you feed the client ID to a window variable called window.ClientId. and reference it in line 114 as 

`var clientId = window.ClientId;`

Provide a secrets.sample.js file in the code with an empty value in the window.ClientId variable and offer documentation that states that the user should save this file as secrets.js and input the client id received on app registration. 

This helps with security

File: PeoplePicker.html (hunk 1 above)
Comment (titusgicheru) [reply PeoplePicker.html]: @thewahome the requested changes above are now effected.

**Review by thewahome-zz — APPROVED** [review]

**Review by MIchaelMainer — CHANGES_REQUESTED** [review]
.vs/ should be set in a .gitignore. These files are typically not checked in to source control.

**Comment by MIchaelMainer** [comment]
@titusgicheru I was late and submitted this review after you merged. We shouldn't be checking in /.vs files.

File: PeoplePicker.html [hunk 3]
```diff
@@ -1,230 +1,183 @@
...
+          </div>        
+          <!-- Create a People Picker from the HTML elements -->
+          <script type="text/javascript">
+            var PeoplePickerElements = document.querySelectorAll(".ms-PeoplePicker");
+            for (var i = 0; i < PeoplePickerElements.length; i++) {
+              new fabric['PeoplePicker'](PeoplePickerElements[i]);
+            }
+          </script>       
+        
+          <!-- Authenticate with Office JS Helper -->
```
Comment (MIchaelMainer) [inline PeoplePicker.html:113]: We are no longer using Office JS Helper

File: PeoplePicker.html [hunk 4]
```diff
@@ -1,230 +1,183 @@
...
+        
+          <!-- Fabric core -->
+          <link rel="stylesheet" href="fabric.min.css">
+          <link rel="stylesheet" href="fabric.components.min.css">
+          <script src="fabric.min.js"></script>
+
+          <!-- Microsoft Graph JavaScript SDK -->                
+          <script type="text/javascript" src="msgraph-sdk-javascript/lib/graph-js-sdk-web.js"></script> 
+
+          <!-- Office JavaScript API Helpers (via CDN) -->
```
Comment (MIchaelMainer) [inline PeoplePicker.html:22]: Remove reference to OfficeJS helper

File: .vs/ProjectSettings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (MIchaelMainer) [inline .vs/ProjectSettings.json:1]: Remove all .vs/ files and update gitignore to not include this directory

File: README.md [hunk 6]
```diff
@@ -28,36 +28,31 @@ This sample requires the following:  
...
-5. Under **Platforms**, choose **Add Platform**.
-
-6. Choose **Web**.
-
-7. Make sure the **Allow Implicit Flow** check box is selected, and enter *https://your host/PeoplePicker.html* as the Redirect URI. 
-
-   The **Allow Implicit Flow** option enables the hybrid flow. During authentication, this enables the app to receive both sign-in info (the id_token) and artifacts (in this case, an authorization code) that the app can use to obtain an access token.
-
-8. Choose **Save**.
+1. Navigate to the [Azure portal - App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page.
```
Comment (MIchaelMainer) [inline README.md:31]: Update the entire readme to reflect the changes. This sample no longer uses OfficeJS [helper](https://github.com/microsoftgraph/javascript-officeuifabric-peoplepicker-sample/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R17).

**Pull request merged**